src/toolbars/TimeSignatureToolBar.cpp

```diff
@@ -0,0 +1,286 @@
+
+   tempoSizer->Add(mLowerSignatureControl, 0, wxEXPAND | wxRIGHT, 5);
+
+   mTempoControl->Bind(
+      wxEVT_SPINCTRL,
+      [this](auto)
+      {
+         const auto tempo = mTempoControl->GetValue();
+         
+         ProjectTimeSignature::Get(mProject).SetTempo(tempo);
+
+         ProjectHistory::Get(mProject).PushState(
+            XO("Tempo Changed"), XO("Tempo Changed"), UndoPush::CONSOLIDATE);
+      });
+
+   mUpperSignatureControl->Bind(
+      wxEVT_SPINCTRL,
+      [this](auto)
+      {
+         const auto upper = int(mUpperSignatureControl->GetValue());
+
+         ProjectTimeSignature::Get(mProject).SetUpperTimeSignature(upper);
+
+         ProjectHistory::Get(mProject).PushState(
+            XO("Upper Time Signature Changed"),
+            XO("Upper Time Signature Changed"), UndoPush::CONSOLIDATE);
+      });
+
+   mLowerSignatureControl->Bind(
+      wxEVT_COMBOBOX,
+      [this](auto)
+      {
+         long value;
+         
+         if (!mLowerSignatureControl->GetValue().ToLong(&value))
+            return;
+
+         ProjectTimeSignature::Get(mProject).SetLowerTimeSignature(value);
+
+         ProjectHistory::Get(mProject).PushState(
+            XO("Lower Time Signature Changed"),
+            XO("Lower Time Signature Changed"), UndoPush::CONSOLIDATE);
+      });
+
+#if wxUSE_ACCESSIBILITY
+   mTempoControl->SetAccessible(safenew WindowAccessible(mTempoControl));
+   mUpperSignatureControl->SetAccessible(
+      safenew WindowAccessible(mUpperSignatureControl));
+   mLowerSignatureControl->SetAccessible(
+      safenew WindowAccessible(mLowerSignatureControl));
+#endif
+   
+   RegenerateTooltips();
+   Fit();
+   Layout();
+}
+
+void TimeSignatureToolBar::UpdatePrefs()
+{
+   // Set label to pull in language change
+   SetLabel(XO("Time Signature"));
+
+   RegenerateTooltips();
+   // Give base class a chance
+   ToolBar::UpdatePrefs();
+}
+
+void TimeSignatureToolBar::RegenerateTooltips()
+{
+}
+
+void TimeSignatureToolBar::OnSize(wxSizeEvent& evt)
+{
+   Refresh( true );
+
+   evt.Skip();
+}
+
+void TimeSignatureToolBar::AddTitle(
+   const TranslatableString& Title, wxSizer* pSizer, int flags, int border,
+   double fontMultiplier)
+{
+   const auto translated = Title.Translation();
+   
+   auStaticText* pTitle = safenew auStaticText(this, translated);
+   
+   pTitle->SetBackgroundColour(theTheme.Colour(clrMedium));
+   pTitle->SetForegroundColour(theTheme.Colour(clrTrackPanelText));
+   pTitle->ScaleFont(fontMultiplier);
+
+   pSizer->Add(pTitle, 0, flags, border);
+}
+
+static RegisteredToolbarFactory factory{
+   []( AudacityProject &project ){
+   return ToolBar::Holder { safenew TimeSignatureToolBar { project } };
+} };
+
+namespace {
+AttachedToolBarMenuItem sAttachment{
+   /* i18n-hint: Clicking this menu item shows the toolbar
+      for selecting a time range of audio */
+   TimeSignatureToolBar::ID(), wxT("ShowTimeSignatureTB"), XXO("Time Signature Toolbar (Beta)")
```

Comment:
Are we really making that string visible to the end user?

Reply:
You mean Beta? Yes, this is by design